Add GET User endpoint

neYaANK opened the pull request:
Add GET User endpoint and test it. Also refactor tests a bit

neYaANK committed:
add and test GET User endpoint, minor changes

Get User endpoint was added for accessing its data without a password.
It is also tested. Some tests are refactored a bit.

amitev commented on Task2/src/main/java/org/neyaank/task2/user/UserServiceImpl.java:
```diff
@@ -32,4 +37,13 @@ public User updateUser(int id, User user) {
+    public User findUserById(int id) {
+        Optional<User> user = userRepository.findById(id);
+        if(user.isEmpty()){
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "User not found");
```
Service-Klassen sollte nichts über die Controllers wissen.

neYaANK replied:
Jetzt Service einfach ein Exception ohne Controllerinformation und nur mit wichtige für Exception Dateien erstellt und dann diese Exception ist in ExceptionHandler behandelt. Ich glaube das sollte passen

neYaANK committed:
stop exposing Password in UserService

amitev commented on Task2/src/test/java/org/neyaank/task2/user/UserTest.java:
```diff
@@ -65,90 +65,91 @@ public void should_returnNewUser_whenRegisterValidUser() throws Exception {
             {"", "Pass$1", "passworddd", "PASSDWORDDD", "$$$$$$$$$$", "Password12",
                     "Password$$", "password$12"})
-    public void should_returnBadRequest_whenInvalidPassword(String value)
+    public void then_returnBadRequest_whenInvalidPassword(String value)
```
Given/When/Then werden prinzipiell als Testschritte verwendet, nicht als Testszenarios.
Der vorige Name war eigentlich okay.

neYaANK committed:
replace exception in UserService to more abstract

Old exception was created directly with NotFound status code in the
service layer, and the service layer should not take care of response
 status codes.
Return the old names for test helper methods.
